### PR TITLE
Configuration and Connection

### PR DESCRIPTION
**Pull request by raphael-goetz**
Adding classes for configuration and connection to redis and rabbitmq

### CODE REVIEW

**Review by Knerio — APPROVED** [review]

File: src/main/java/tech/code0/communication/RabbitConnection.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            return connection;
+        } catch (Exception exception) {
+            LOGGER.warning("Connection to RabbitMQ failed: " + exception.getMessage());
+            throw new RuntimeException(exception);
+        }
+    }
+
+    public void close() {
+        try {
+            LOGGER.warning("Closing RabbitMQ connection");
```
Comment (Taucher2003) [inline src/main/java/tech/code0/communication/RabbitConnection.java:36]: Closing connections is a standard procedure during an application lifecycle, so this is not really a warning
```suggestion
            LOGGER.info("Closing RabbitMQ connection");
```

File: src/main/java/tech/code0/data/RedisConnection.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    public RedisConnection(AquilaConfiguration configuration) {
+        LOGGER.info("Initializing Redis connection");
+        this.connectionString = "redis://:flows@" + configuration.getRedisHost() + ":" + configuration.getRedisPort();
+        this.client = RedisClient.create(connectionString);
+        this.connection = client.connect();
+        LOGGER.info("Connected to Redis");
+    }
+
+    public void shutdown() {
+        LOGGER.warning("Shutting down RedisConnection");
```
Comment (Taucher2003) [inline src/main/java/tech/code0/data/RedisConnection.java:26]:
```suggestion
        LOGGER.info("Shutting down RedisConnection");
```

**Review by Taucher2003 — CHANGES_REQUESTED** [review]

File: src/main/java/tech/code0/configuration/AquilaConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        this.redisHost = getEnvVar("RABBITMQHOST");
+
+        this.rabbitMQPort = Integer.parseInt(getEnvVar("RABBITMQPORT"));
+        this.redisPort = Integer.parseInt(getEnvVar("RABBITMQPORT"));
+
+        final var port = System.getenv("BACKENDPORT");
+        this.backendPort = (port != null) ? Integer.parseInt(port) : 0;
+    }
+
+    private String getEnvVar(String varName) {
```
[the remark this answers is not in the document]
Comment (raphael-goetz) [reply src/main/java/tech/code0/configuration/AquilaConfiguration.java:33]: We didn't come to a conclusion to a default port yesterday. Just put a placeholder there.

File: src/main/java/tech/code0/util/AquilaLogger.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package tech.code0.util;
+
+import lombok.experimental.UtilityClass;
+
+import java.util.logging.Logger;
+
+@UtilityClass
+public class AquilaLogger {
+
+    public static final Logger LOGGER = Logger.getLogger(AquilaLogger.class.getName());
```
[the remark this answers is not in the document]
Comment (raphael-goetz) [reply src/main/java/tech/code0/util/AquilaLogger.java]: Changed the global logger to a local one. If you could elaborate the features of such logger I'll implement it asap

**Review by Taucher2003 — APPROVED** [review]

**Pull request merged**